### PR TITLE
SIGINT-1663,  SIGINT-1291

### PR DESCRIPTION
**Pull request by maksudur-rahman-maruf**
- Avoided passing empty values to synopsys bridge, if the values are not set
- Added new Polaris parameters polaris_prComment_enabled, polaris_prComment_severities, polaris_branch_parent_name
- Added the above parameters in the pipeline syntax UI.
- Integrated all the supported scms Github, Gitlab and Bitbucket
- Added unit test cases for Polaris PR Comment

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/service/scan/blackduck/BlackDuckParametersService.java [hunk 1]
```diff
@@ -86,43 +87,55 @@ public BlackDuck prepareBlackDuckObjectForBridge(Map<String, Object> blackDuckPa
...
+                blackDuck.getScan().getFailure().setSeverities(failureSeverities);
+            }
         }
     }
 
-    private void setAutomationFixpr(BlackDuck blackDuck, String value) {
+    private void setAutomationFixpr(BlackDuck blackDuck, String value, Automation automation) {
         if (isBoolean(value)) {
+            blackDuck.setAutomation(automation);
             blackDuck.getAutomation().setFixpr(Boolean.parseBoolean(value));
```
Comment (zaman-akib) [inline src/main/java/io/jenkins/plugins/synopsys/security/scan/service/scan/blackduck/BlackDuckParametersService.java:122]: Can be refactored to - 
```
automation.setFixpr(Boolean.parseBoolean(value));
blackDuck.setAutomation(automation);
```

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/service/scan/blackduck/BlackDuckParametersService.java [hunk 2]
```diff
@@ -86,43 +87,55 @@ public BlackDuck prepareBlackDuckObjectForBridge(Map<String, Object> blackDuckPa
...
         }
     }
 
-    private void setAutomationPrComment(BlackDuck blackDuck, String value) {
+    private void setAutomationPrComment(BlackDuck blackDuck, String value, Automation automation) {
         if (isBoolean(value)) {
+            blackDuck.setAutomation(automation);
             blackDuck.getAutomation().setPrComment(Boolean.parseBoolean(value));
```
Comment (zaman-akib) [inline src/main/java/io/jenkins/plugins/synopsys/security/scan/service/scan/blackduck/BlackDuckParametersService.java:129]: Can be refactored to - 
```
automation.setPrComment(Boolean.parseBoolean(value));
blackDuck.setAutomation(automation);
```

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/service/scan/blackduck/BlackDuckParametersService.java [hunk 3]
```diff
@@ -86,43 +87,55 @@ public BlackDuck prepareBlackDuckObjectForBridge(Map<String, Object> blackDuckPa
...
             String[] failureSeveritiesInput = value.toUpperCase().split(",");
 
             for (String input : failureSeveritiesInput) {
                 failureSeverities.add(input.trim());
             }
-            blackDuck.getScan().getFailure().setSeverities(failureSeverities);
+            if (!failureSeverities.isEmpty()) {
+                blackDuck.setScan(scan);
+                blackDuck.getScan().setFailure(new Failure());
+                blackDuck.getScan().getFailure().setSeverities(failureSeverities);
```
Comment (zaman-akib) [inline src/main/java/io/jenkins/plugins/synopsys/security/scan/service/scan/blackduck/BlackDuckParametersService.java:114]: Can be refactored to - 
```
Failure failure = new Failure();
failure.setSeverities(failureSeverities);
scan.setFailure(failure);
blackDuck.setScan(scan);
```

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/service/scan/coverity/CoverityParametersService.java [hunk 4]
```diff
@@ -110,13 +111,20 @@ public Coverity prepareCoverityObjectForBridge(Map<String, Object> coverityParam
                     coverity.getConnect().getStream().setName(value);
                     break;
                 case ApplicationConstants.COVERITY_POLICY_VIEW_KEY:
-                    coverity.getConnect().getPolicy().setView(value);
+                    if (!value.isBlank()) {
+                        coverity.getConnect().setPolicy(new Policy());
+                        coverity.getConnect().getPolicy().setView(value);
```
Comment (zaman-akib) [inline src/main/java/io/jenkins/plugins/synopsys/security/scan/service/scan/coverity/CoverityParametersService.java:116]: refactor this to have the consistency, also check the below `cases`

**Review by zaman-akib — CHANGES_REQUESTED** [review]

**Review by zaman-akib — APPROVED** [review]

**Review by jahid1209 — APPROVED** [review]

**Pull request merged**